cupyx/scipy/sparse/compressed.py

```diff
@@ -130,7 +130,9 @@ def __init__(self, arg1, shape=None, dtype=None, copy=False):
                              % (len(indptr), major + 1))
 
         self._descr = cusparse.MatDescriptor.create()
-        self._shape = shape
+        if not util.isshape(shape):
+            raise ValueError('invalid shape (must be a 2-tuple of int)')
+        self._shape = int(shape[0]), int(shape[1])
```

Comment:
If it is the case that this is only relevant if `shape` is supplied as an argument (that's not `None`), how about doing this processing at the top?

Reply:
I have moved it up to the top as suggested. Thanks!